3-networks/envs/shared/hierarchical_firewall.tf

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module "hierarchical_firewall_policy" {
+  source = "../../modules/hierarchical_firewall_policy/"
+  parent = data.google_active_folder.common.name
+  name   = "common-firewall-rules"
+  associations = [
+    data.google_active_folder.common.name,
+    data.google_active_folder.bootstrap.name,
+    data.google_active_folder.development.name,
+    data.google_active_folder.production.name,
+    data.google_active_folder.non-production.name,
+  ]
+  rules = {
+    delegate-rfc1918-ingress = {
```

Comment:
What are the purpose of these two delegation rules? Is it to allow all internal traffic in both directions prior to VPC firewall rules being evaluated?

Reply:
These two rules delegate RFC1918 traffic to lower levels (i.e. we're not making any decision in regards to that traffic here).

Again, check the document for the rationale.

Reply:
Ok thanks - just read through it. The doc in 7.6.1 also mentions blocking egress to the internet but I can't see a rule for that here?